src/KeyDB.php

```diff
@@ -52,8 +53,7 @@ public static function createFromRootMetadata(RootMetadata $rootMetadata)
             // One key ID for each $keyMeta['keyid_hash_algorithms'].
             $computedKeyIds = self::computeKeyIds($keyMeta);
             foreach ($computedKeyIds as $keyId) {
-                $keyMeta['keyid'] = $keyId;
-                $db->addKey($keyMeta);
```

Comment:
The only reason why `$keyMeta['id'] was set was so that `addKey()` could use it when setting `$this->keys[$keyId]`. Since $keyMeta is not an object means that all the previous items set using `$this->keys` actual refer to the same object. So the object will have the `keyId` of the last element of `$computedKeyIds`

We don't need to set `keyId` at all and can move the call to `computeKeyIds()` inside `addKey()`. That the process of adding items to `$this->keys` we be encapsulated inside `addKey()`. This is will make added the keys from `targets.json` simpler.